embassy-nrf/src/lib.rs

```diff
@@ -76,6 +76,8 @@ pub mod gpio;
 pub mod gpiote;
 #[cfg(not(any(feature = "_nrf5340", feature = "_nrf9160")))]
 pub mod nvmc;
+#[cfg(feature = "nrf52840")]
```

Comment:
Looking at Nordic infocenter, it looks like PDM should be available on the nRF 52833, 52811, 52810, 5340 and 9160 as well.

Reply:
It seems only available from the application core, which I think you can guard with the feature "_nrf5340-app" (both secure and non-secure mode).

For the name, I think you should use PDM0 as used by the PAC.